### PR TITLE
Adds additional data to PayIr verify result

### PR DESCRIPTION
**Pull request by azhdari**

### CODE REVIEW

File: src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Parbad. All rights reserved.
+// Licensed under the GNU GENERAL PUBLIC License, Version 3.0. See License.txt in the project root for license information.
+
+using Parbad.Gateway.PayIr.Internal;
+
+namespace Parbad.Gateway.PayIr
+{
+    public static class PayIrVerifyAdditionalDataExtensions
+    {
```
Comment (Sina-Soltani) [inline src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs:9]: Add a constant here for the additional key.
The key will be used for setting and getting the data.

public const string AdditionalDataKey = "PayIrAdditionalDataKey";

File: src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        public static PayIrVerifyAdditionalData GetPayIrAdditionalData(this IPaymentVerifyResult result)
+        {
+            var additionalData = new PayIrVerifyAdditionalData();
+
+            if (result.AdditionalData.TryGetValue(nameof(PayIrVerifyResponseModel.CardNumber), out var cardNumber))
```
Comment (Sina-Soltani) [inline src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs:14]: No need to set and get the data 4 times! Only 1 time is fine.
return (PayIrVerifyAdditionalData) result.AdditionalData[AdditionalDataKey];

File: src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs [hunk 3]
```diff
@@ -91,7 +91,13 @@ public static PaymentVerifyResult CreateVerifyResult(string response, MessagesOp
                 return PaymentVerifyResult.Failed(message);
             }
 
-            return PaymentVerifyResult.Succeed(result.TransId, messagesOptions.PaymentSucceed);
+            var verifyResult = PaymentVerifyResult.Succeed(result.TransId, messagesOptions.PaymentSucceed);
```
Comment (Sina-Soltani) [inline src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs:94]: Create the PayIrVerifyAdditionalData class here and fill the properties and then add it inside the dictionary.
verifyResult.AdditionalData.Add(YourExtensionsClass.AdditionalDataKey] = class;

File: src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs (hunk 2 above)
Comment (azhdari) [reply src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs:14]: I don't get it. I want to add 4 different additional data.
You mean all those fields will be serialized?

File: src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs (hunk 3 above)
Comment (azhdari) [reply src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs:94]: AdditionalData is a <string,string> dictionary. Should I serialize the class?
What is the harm to add 4 items to the dictionary?

Comment (Sina-Soltani) [reply src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs:94]: That's right. The AdditionalData is still string,string. I didn't commit the new changes yet.
It's going to be a dictionary of <string, object> after I commit the new changes.
So for now let it be like this.

File: src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs (hunk 2 above)
Comment (Sina-Soltani) [reply src/Parbad.Gateway/PayIr/src/PayIrVerifyAdditionalDataExtensions.cs:14]: Yes I serialized the additional data before.
But in new version the AdditionalData is a dictionary of <string, object>.
I explained it bellow.
No need to touch it for now.

**Comment by Sina-Soltani** [comment]
You forgot to update the nuget version.
Change it to 1.1.0.
You should also add a release note so the users can understand what is changed in the new version. 
```
<PackageReleaseNotes>Additional Data are now included in the result object.</PackageReleaseNotes>
```

**Comment by Sina-Soltani** [comment]
Nice work.

**Pull request merged**